File: src/main/java/com/cuupa/classificator/to/Document.java
package com.cuupa.classificator.to;


import java.io.Serializable;

public class Document implements Serializable {
	
	private static final long serialVersionUID = 1L;
	
	private byte[] content;
	
	private String filename;

	private boolean encrypted;

	public void setContent(byte[] content) {
		this.content = content;
	}

	public void setFileName(String filename) {
		this.filename = filename;
	}

	public byte[] getContent() {
		return content;
	}

	public String getFileName() {
		return filename;
	}

	public void setEncrypted(boolean encrypted) {
		this.encrypted = encrypted;
	}

	public boolean isEncrypted() {
		return encrypted;
	}

}

File: src/main/java/com/cuupa/classificator/to/Document.kt
package com.cuupa.classificator.to

import java.io.Serializable

data class Document(val content: ByteArray, var filename: String?, var encrypted: Boolean) : Serializable {

    companion object {
        private const val serialVersionUID = 1L
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Document

        if (!content.contentEquals(other.content)) return false
        if (filename != other.filename) return false
        if (encrypted != other.encrypted) return false

        return true
    }

    override fun hashCode(): Int {
        var result = content.contentHashCode()
        result = 31 * result + (filename?.hashCode() ?: 0)
        result = 31 * result + encrypted.hashCode()
        return result
    }
}